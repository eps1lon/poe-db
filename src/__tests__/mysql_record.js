const fs = require('fs');
const path = require('path');

const MysqlRecord = require('../MysqlRecord');

const records = JSON.parse(
  fs.readFileSync(path.join(__dirname, './records.json')),
);

const props = {};
records.forEach(record => {
  props[record['$']['file']] = record;
});

describe('MysqlRecord', () => {
  describe('Table', () => {
    const ascendancy = new MysqlRecord(props['Ascendancy'], records);

    it('should have lower case underscore table names', () => {
      expect(ascendancy.tableName()).toBe('ascendancy');
    });
  });

  describe('cols', () => {
    it('should build a cols object', () => {
      const ascendancy = new MysqlRecord(props['Ascendancy'], records);

      expect(ascendancy.cols()).toMatchSnapshot();
    });
  });
});
