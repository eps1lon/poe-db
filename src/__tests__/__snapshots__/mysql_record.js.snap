// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MysqlRecord cols should build a cols object 1`] = `
Array [
  Object {
    "name": "id",
    "type": "BIGINT UNSIGNED",
  },
  Object {
    "name": "class_no",
    "type": "INT",
  },
  Object {
    "name": "characters_id",
    "type": "BIGINT UNSIGNED",
  },
  Object {
    "name": "coordinate_rect",
    "type": "TEXT",
  },
  Object {
    "name": "rgb_flavour_text_colour",
    "type": "TEXT",
  },
  Object {
    "name": "name",
    "type": "TEXT",
  },
  Object {
    "name": "flavour_text",
    "type": "TEXT",
  },
  Object {
    "name": "ogg_file",
    "type": "TEXT",
  },
]
`;
